client/src_editor/Editor_Blocks.re

```diff
@@ -139,18 +166,134 @@ let make =
         | Block_FocusUp(_)
         | Block_FocusDown(_)
         | Block_FocusNextBlockOrCreate
+        | Block_CaptureQueuedMeta(_, _, _)
+        | Block_QueueDelete(_)
+        | Block_RefmtAsLang(_)
+        | Block_CleanBlocksCopy
+        | Block_RefmtOnDemand
         | Block_Execute(_) => ()
         | Block_Add(_, _)
-        | Block_QueueDelete(_)
         | Block_DeleteQueued(_)
         | Block_Restore(_)
-        | Block_CaptureQueuedMeta(_, _, _)
-        | Block_UpdateValue(_, _, _) => onUpdate(newSelf.state.blocks)
+        | Block_MapRefmtToBlocks(_, _, _) => onUpdate(newSelf.state.blocks)
+        | Block_UpdateValue(blockId, newValue, diff) =>
+          /* switch(newSelf.state.blocksCopy) {
+               | None => ()
+               | Some(_) =>
+                 if (diff.origin == "+input") {
```

Comment:
It's diff -> CodeMirror.Diff.originGet 
but could you make this a black list insead of a white list? (only filter out unneeded event)

Reply:
I'm not sure what you mean... do you mean use a catch all instead of listing out the actions we're not reacting to? e.g. `_ => ()`

Reply:
I'm not sure what you mean... do you mean use a catch all instead of listing out the actions we're not reacting to? e.g. `_ => ()`

Reply:
I think we can filter out setValue origin and not triggering onUpdate event